# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmGC.cpp
# Commit: cf96078694ae
# Full Hash: cf96078694aea74cae4557f683a76c6de6ba9cf1
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 3
# Description:
#   Bug 1692065 - wasm: Rename MIRType::RefOrNull to 'WasmAnyRef'. r=yury
#   
#   This is a mechanical change. The previous name 'RefOrNull' was too
#   generic, and it wasn't clear that the type was actually a wasm::AnyRef
#   type.
# ==============================================================================

diff -r 662cb63c42d7 -r cf96078694ae js/src/wasm/WasmGC.cpp
--- a/js/src/wasm/WasmGC.cpp	Fri Aug 04 14:06:49 2023 +0000
+++ b/js/src/wasm/WasmGC.cpp	Fri Aug 04 14:06:49 2023 +0000
@@ -132,7 +132,7 @@
   for (WasmABIArgIter i(argTypes); !i.done(); i++) {
     ABIArg argLoc = *i;
     if (argLoc.kind() == ABIArg::Stack &&
-        argTypes[i.index()] == MIRType::RefOrNull) {
+        argTypes[i.index()] == MIRType::WasmAnyRef) {
       uint32_t offset = argLoc.offsetFromArgBase();
       MOZ_ASSERT(offset < nInboundStackArgBytes);
       MOZ_ASSERT(offset % sizeof(void*) == 0);
@@ -180,7 +180,7 @@
   }
 
   for (WasmABIArgIter i(args); !i.done(); i++) {
-    if (!i->argInRegister() || i.mirType() != MIRType::RefOrNull) {
+    if (!i->argInRegister() || i.mirType() != MIRType::WasmAnyRef) {
       continue;
     }
 